src/Confluent.Kafka/Producer.cs

```diff
@@ -194,6 +194,150 @@ public void HandleDeliveryReport(Message message)
                 SetResult(message);
             }
         }
+        
+        /// <summary>
+        ///     Detailed messaged
+        /// </summary>
+        public struct ProduceRecord
```

Comment:
interesting point. ok, I'll think about this a bit more before having an opinion.